index/generator/schema/schema.go

```diff
@@ -88,6 +88,17 @@ type Schema struct {
 	Git               *Git                   `yaml:"git,omitempty" json:"git,omitempty"`
 }
 
+// IndexType describes the type of index entry
+type IndexType string
```

Comment:
I think the type should of `DevfileType` as currently we support sample devfile and stack devfile. We also need to update the comment to make it more clear as the index entry doesn't have type.

I assume the type will be used on dev console side and that's why we export it, otherwise we can make the type local.

Reply:
Makes sense, updated!

Schema structs should remain exported because it allows access to others much like devfile/api schema structs.